### PR TITLE
Add error filter

### PR DESCRIPTION
**Pull request by Angel-M-R**

### CODE REVIEW

File: app/instance-initializers/bugsnag.js [hunk 1]
```diff
@@ -23,7 +24,8 @@ export function initialize(instance) {
 		setProperties(this, {
 			owner,
 			router,
-			client
+			client,
+			errorFilter
```
Comment (adrigzr) [inline app/instance-initializers/bugsnag.js]: This is not necessary. You can use `appMethods.filterError`

File: app/instance-initializers/bugsnag.js [hunk 2]
```diff
@@ -42,6 +44,15 @@ export default {
 	},
 
 	_onError(error) {
+
+		if (!error || this.errorFilter.includes(error.name)) {
```
Comment (adrigzr) [inline app/instance-initializers/bugsnag.js]: This is not good, you should call the function directly.

```suggestion
		if (!error || !appMethods.errorFilter(error)) {
```

File: app/instance-initializers/bugsnag.js [hunk 3]
```diff
@@ -42,6 +44,15 @@ export default {
...
+			return;
+		}
+
+		if (typeof error !== 'object' || !error.name || !error.message) {
```
Comment (adrigzr) [inline app/instance-initializers/bugsnag.js]: This is not longer needed. It was removed on `master`.

File: tests/acceptance/bugsnag-test.js [hunk 4]
```diff
@@ -141,4 +143,30 @@ module('Acceptance | bugsnag', (hooks) => {
 		// eslint-disable-next-line require-atomic-updates
 		appMethods.getMetaData = getMetaData;
 	});
+
+	test('it does not notify a filtered error', async function(assert) {
```
Comment (adrigzr) [inline tests/acceptance/bugsnag-test.js:145]: This test is not working properly. You need to stub de `errorFilter` method.

**Review by adrigzr — CHANGES_REQUESTED** [review]

**Review by adrigzr — APPROVED** [review]

**Review by eumendoza — APPROVED** [review]

**Pull request merged**